lib/fluent/plugin/in_tail.rb

```diff
@@ -251,7 +267,11 @@ def close_watcher_after_rotate_wait(tw)
     def flush_buffer(tw)
       if lb = tw.line_buffer
         lb.chomp!
-        lb.force_encoding(@encoding) if @encoding
+        if @encoding && @from_encoding
```

Comment:
Here is microbenchmark result. I think default should be fast. So `f2` implementation is better for almost users.

```
% ruby b_enc_perf.rb
Rehearsal --------------------------------------------------------
f1 with nil / nil      0.840000   0.000000   0.840000 (  0.842853)
f2 with nil / nil      0.760000   0.000000   0.760000 (  0.769146)
f1 with true / true    0.870000   0.000000   0.870000 (  0.871896)
f2 with true / true    0.920000   0.000000   0.920000 (  0.917469)
f1 with true / false   0.990000   0.000000   0.990000 (  0.989552)
f2 with true / false   0.920000   0.010000   0.930000 (  0.926458)
----------------------------------------------- total: 5.310000sec

                           user     system      total        real
f1 with nil / nil      0.890000   0.000000   0.890000 (  0.890333)
f2 with nil / nil      0.820000   0.000000   0.820000 (  0.822715)
f1 with true / true    0.890000   0.000000   0.890000 (  0.885648)
f2 with true / true    0.940000   0.000000   0.940000 (  0.938831)
f1 with true / false   1.000000   0.000000   1.000000 (  1.003493)
f2 with true / false   0.950000   0.000000   0.950000 (  0.956331)
```

Code:

```rb
require 'benchmark'

class Foo
  def initialize(enc, from_enc)
    @enc = enc
    @from_enc = from_enc
  end

  def f1(num)
    num.times {
      if @enc && @from_enc
        1
      elsif @enc
        2
      end
    }
  end

  def f2(num)
    num.times {
      if @enc
        if @from_enc
          1
        else
          2
        end
      end
    }
  end
end

n = 20000000

Benchmark.bmbm do |x|
  x.report('f1 with nil / nil') {
    foo = Foo.new(nil, nil)
    foo.f1(n)
  }
  x.report('f2 with nil / nil') {
    foo = Foo.new(nil, nil)
    foo.f2(n)
  }

  x.report('f1 with true / true') {
    foo = Foo.new(true, true)
    foo.f1(n)
  }
  x.report('f2 with true / true') {
    foo = Foo.new(true, true)
    foo.f2(n)
  }

  x.report('f1 with true / false') {
    foo = Foo.new(true, false)
    foo.f1(n)
  }
  x.report('f2 with true / false') {
    foo = Foo.new(true, false)
    foo.f2(n)
  }
end
```